### ISSUE TITLE
iOS Release Crash: Flutter News Toolkit

### ISSUE THREAD

**Issue by felangel** [shorebirdtech/shorebird#1702]
Running the base release of https://github.com/flutter/news_toolkit on iOS  results in the following crash:

```
*** First throw call stack:
(0x1b35f1a24 0x1ac9e1958 0x1b374a980 0x1030db3e8 0x1030db55c 0x1ba507850 0x1ba5087c8 0x1ba4dfad8 0x1ba4ecd48 0x1ba4ed514 0x1fc05cb14 0x1fc05c67c)
* thread #8, queue = 'com.apple.root.default-qos', stop reason = signal SIGABRT
    frame #0: 0x00000001ecf52274 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`:
->  0x1ecf52274 <+8>:  b.lo   0x1ecf52290               ; <+36>
    0x1ecf52278 <+12>: stp    x29, x30, [sp, #-0x10]!
    0x1ecf5227c <+16>: mov    x29, sp
    0x1ecf52280 <+20>: bl     0x1ecf4dc54               ; cerror_nocancel
Target 0: (Runner) stopped.
```

**Comment by felangel** [comment]
Update: This seems like a false alarm. The app is crashing even without Shorebird investigating further...

**Comment by felangel** [comment]
Update 2: Nevermind this is actually looking like a real crasher

```
* thread #11, name = 'io.flutter.1.ui', stop reason = EXC_BAD_ACCESS (code=1, address=0x3f80000000000000)
    frame #0: 0x00000001083acc78 Flutter`___lldb_unnamed_symbol51617 + 180
Flutter`___lldb_unnamed_symbol51617:
->  0x1083acc78 <+180>: ldr    x19, [x19]
    0x1083acc7c <+184>: adrp   x8, 1181
    0x1083acc80 <+188>: ldr    x20, [x8, #0x3e0]
    0x1083acc84 <+192>: str    x21, [sp, #0x8]
Target 0: (Runner) stopped.
```

**Comment by eseidel** [comment]
We've found the bug and have a partial fix, working to finish the fix and release today.

**Comment by eseidel** [comment]
This ended up being a series of fixes to our ffi handling code.  We weren't correctly handling arguments beyond the 7th, or struct returns.  It was only broken in the CPU -> FFI -> CPU case which is why it just showed up in our last release, where we turned on "run on the cpu" by default for release builds.

Will be fixed in our next release in about 30 minutes.

**Comment by eseidel** [comment]
This fix was released in 0.25.2 (or rather the Flutter version that release defaults to): https://github.com/shorebirdtech/shorebird/blob/main/RELEASE_NOTES.md#0252-february-7-2024
